packages/next/next-server/server/lib/path-match.ts

```diff
@@ -16,6 +16,20 @@ export default (customRoute = false) => {
       if (!res) {
         return false
       }
+
+      if (customRoute) {
+        const newParams: { [k: string]: string } = {}
+        Object.keys(res.params).forEach((key, idx) => {
+          if (key === idx + '') {
```

Comment:
This will break if a user custom defines a key as `/:0` -- the regex keys are always inserted as ints so this should probably be a strict match (note `'2'` vs `0`).

```js
> p.pathToRegexp('/:2/(.*)', keys)
/^(?:\/([^\/#\?]+?))(?:\/(.*))[\/#\?]?$/i
> keys
[
  {
    name: '2',
    prefix: '/',
    suffix: '',
    pattern: '[^\\/#\\?]+?',
    modifier: ''
  },
  { name: 0, prefix: '/', suffix: '', pattern: '.*', modifier: '' }
]
```

Reply:
Updated to be a strict match and added a test for supporting named params that look like unnamed params